openapi/geonames/client.bal

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Provides API key configurations needed when communicating with a remote HTTP endpoint.
+public type ApiKeysConfig record {|
+    # Represents API Key `username`
+    string username;
+|};
+
+# This is a generated connector for [Geonames API](https://www.geonames.org/export/JSON-webservices.html) OpenAPI specification.
+# Geonames is a geographical database where users can retrieve geo location information.
+@display {label: "Geonames", iconPath: "icon.png"}
+public isolated client class Client {
+    final http:Client clientEp;
+    final readonly & ApiKeysConfig apiKeyConfig;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials.
+    # Create a [Geonames account](https://www.geonames.org/login) and obtain tokens by following [this guide](https://www.geonames.org/export/web-services.html). 
+    #
+    # + apiKeyConfig - API keys for authorization 
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ApiKeysConfig apiKeyConfig, http:ClientConfiguration clientConfig =  {}, string serviceUrl = "api.geonames.org/") returns error? {
```

Comment:
Shall we add a valid default service URL here

Reply:
Fixed in https://github.com/ballerina-platform/openapi-connectors/commit/410e9f41506c94f4a229ce0eb8e2634606237f26